src/appliance-onboarding-script/appliance_setup/avs/avsarconboarder/orchestrator/network/_network_orchestrator.py

```diff
@@ -0,0 +1,57 @@
+from ...constants import Constant
+from ...orchestrator._orchestrator import Orchestrator
+import ipaddress
+from avs.avsarconboarder.exception import InvalidInputError
+
+class NetworkOrchestrator(Orchestrator):
+
+    def validate_ip_address(self, ip_addr):
+        try:
+            ipaddress.ip_address(ip_addr)
+        except:
+            raise InvalidInputError("Invalid ip addresses provided")
+
+    def validate_static_ip_cidr_block(self, segment_ip_cidr, version):
+        gateway_ip, cidr = self.get_gateway_address_cidr_from_network_addr(segment_ip_cidr)
+        self.validate_ip_address(gateway_ip)
+        
+        if version == Constant.NEW_CONFIG_VERSION:
+            if int(cidr) != 28:
+                raise InvalidInputError("Invalid segment block size provided, Please provide a /28 address")
+            
+            subnet_first_ip_addr = ipaddress.IPv4Address(gateway_ip) - 1
+            try:
+                ipaddress.ip_network('{}/{}'.format(subnet_first_ip_addr,cidr))
+            except:
+                raise InvalidInputError("Invalid gateway ip provided, please provide a segment with first gateway ip")          
+        
+    def get_gateway_address_cidr_from_network_addr(self, segment_ip_cidr):
+        char_index = 0
+        gateway_ip = ""
+        while char_index < len(segment_ip_cidr) and segment_ip_cidr[char_index] != '/':
+            gateway_ip += segment_ip_cidr[char_index]
+            char_index += 1
+
+        cidr = segment_ip_cidr[char_index - len(segment_ip_cidr) + 1:]
+
+        return gateway_ip, cidr
+
+    '''
+    gateway ip = 2nd ip of cidr = 10.0.0.(xxxx 0001)
+    control plane IP address = 3rd ip of cidr = 10.0.0.(xxxx 0002)
+    nodepool range = 11th to 15th ip = 10.0.0.(xxxx 1010) - 10.0.0.(xxxx 1110)  
```

Comment:
>xxxx 1010 [](http://example.com/codeflow?start=47&length=9)

is this binary notation?  the above seems to be decimal. may be i am missing something. let us use a single notation ( decimal is more readable in my opinion, but either is fine)

Reply:
let me use a decimal system.